File: src/EnglishGame/Models/Workbook/UExercise.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace EnglishGame.Models.Workbook
{
    public class UExercise
    {
        public int Id;
        public string Name;
        public UTeacher Teacher;
        public List<UTask> Tasks;
    }
}

File: src/EnglishGame/wwwroot/app/workbook/models/uexercise.ts
import { UTask } from './utask';
import { UPupil } from './upupil';
import { UTeacher } from './uteacher';
import { UScheduleItem } from './uscheduleitem';

export class UExercise {
    id: number;
    name: string;
    teacher_id: number;
    taskList: UTask[];
    pupilList: UPupil[];
    scheduleItemList: UScheduleItem[];
    teacher: UTeacher;
}